providers/gov/usgs/harvester.py

```diff
@@ -1,5 +1,42 @@
+from furl import furl
+
+from datetime import date, timedelta
+
 from share.harvest.harvester import Harvester
 
 
 class USGSHarvester(Harvester):
-    pass
+    url = 'https://pubs.er.usgs.gov/pubs-services/publication'
+
+    def do_harvest(self, start_date, end_date):
+        today = date.today()
+        end_date = end_date.date()
+        start_date = start_date.date()
+
+        end_days_back = (today - end_date).days
+        start_days_back = (today - start_date).days
+
+        # The USGS API does not support date ranges
+        for days_back in range(end_days_back, start_days_back):
+            page = 1
+            resp = self.requests.get(furl(self.url).set(query_params={
```

Comment:
Sorry to be nitpicky here, but could you refactor this to be a bit more DRY?
It's kinda sucky to do because of python's lack of do while loops but it leads to better maintainability. 
[This SO post](https://stackoverflow.com/questions/743164/emulate-a-do-while-loop-in-python) is the best way I've come across to get psuedo do whiles